### DISCUSSION TITLE
Feature Request: More Zoom!

### DISCUSSION THREAD

**Discussion by chrisl8** [Freika/dawarich#260] [Ideas]
I would like either the maximum zoom to be increased or to be configurable. I often want to zoom in further than allowed, but the max is hard coded to 19. I'm not sure what it should be, maybe that is a good default, but I have been setting it to 25 myself.

**Comment by Freika** [comment]
Not all maps layers support closer zoom (read: has that detailed tiles) but I'll have look if I can increase it for some of them

**Comment by chrisl8** [comment]
I don't think it matters if the map itself doesn't support the zoom level, it just goes to a blank field. The point is to zoom in enough on points to be able to differentiate them and click on them.

**Reply by Freika** [reply]
It's not entirely correct. Here is an example: 


https://github.com/user-attachments/assets/80ebfb1e-e12d-4dcc-a892-355d152bf7ec



Basically, if you zoom close enough, and then switch to a layer that does not support this zoom level, Leaflet won't allow you to switch layers. And if you manage to switch it anyway, it will show you a gray background instead of map tiles, because there are none for this level of zoom.

**Reply by chrisl8** [reply]
Ah, interesting. I don't think I tried zoom 25 when all of those map options existed. I'll go back and try it again and see what results I get.

**Comment by chrisl8** [comment]
Here is an example of what I'm referring to. At the default max zoom level of 19, I cannot get close enough to find the point that is making that crazy line come in from the upper left. The points are just overlapping.

![image](https://github.com/user-attachments/assets/e02c7552-0dc2-42b3-9ef5-714ea92712d9)

If I set the `maxZoom` to 24 though, I can zoom right in there and find it.
https://github.com/Freika/dawarich/blob/e755d21e79e9c367e88f6b452210fe18f93e5810/app/javascript/maps/layers.js#L7

![image](https://github.com/user-attachments/assets/6eb369e1-fcd1-4c54-b8df-a10395374722)

I see what you mean though about how it breaks switching between maps.

**However, you already have this issue in the existing code. I see you have the max zoom set to different levels for the different maps. So, for example, if I zoom in fully on the OPNV map which has `maxZoom` of 18, I cannot select the openTopo map which has `maxZoom` set to 17.**

https://github.com/Freika/dawarich/blob/e755d21e79e9c367e88f6b452210fe18f93e5810/app/javascript/maps/layers.js#L35

https://github.com/Freika/dawarich/blob/e755d21e79e9c367e88f6b452210fe18f93e5810/app/javascript/maps/layers.js#L49

I guess the code needs some more "smarts" to help it to "un-zoom" when switching maps.

**Reply by Freika** [reply]
The video I attached above represents state of the map with `maxZoom` attribute set to 19 for all map layers.

But I got your idea, and I'll see of I can bypass Leaflet's behaviour and zoom closer with the latest tiles even if it would mean having blurry tiles.
